vms/platformvm/state/state.go

```diff
@@ -1689,38 +1660,36 @@ func (s *state) loadPendingValidators() error {
 // Invariant: initValidatorSets requires loadCurrentValidators to have already
 // been called.
 func (s *state) initValidatorSets() error {
-	if s.cfg.Validators.Count(constants.PrimaryNetworkID) != 0 {
-		// Enforce the invariant that the validator set is empty here.
-		return errValidatorSetAlreadyPopulated
-	}
-	err := s.ApplyCurrentValidators(constants.PrimaryNetworkID, s.cfg.Validators)
-	if err != nil {
-		return err
-	}
+	for subnetID, validators := range s.currentStakers.validators {
+		if s.cfg.Validators.Count(subnetID) != 0 {
+			// Enforce the invariant that the validator set is empty here.
+			return fmt.Errorf("%w: %s", errValidatorSetAlreadyPopulated, subnetID)
+		}
 
-	vl := validators.NewLogger(s.ctx.Log, s.bootstrapped, constants.PrimaryNetworkID, s.ctx.NodeID)
-	s.cfg.Validators.RegisterCallbackListener(constants.PrimaryNetworkID, vl)
+		for nodeID, validator := range validators {
+			staker := validator.validator
+			if err := s.cfg.Validators.AddStaker(subnetID, nodeID, staker.PublicKey, staker.TxID, staker.Weight); err != nil {
+				return err
+			}
+
+			delegatorIterator := NewTreeIterator(validator.delegators)
+			for delegatorIterator.Next() {
+				staker := delegatorIterator.Value()
```

Comment:
nit: prefer to give this a different variable name

Reply:
I always end up causing a bug at some point in the future when I do nested for loops with the same var name in them